network/src/node.rs

```diff
@@ -62,7 +63,8 @@ pub struct InnerNode<S: Storage + core::marker::Sync + Send + 'static> {
     /// The list of connected and disconnected peers of this node.
     pub peer_book: PeerBook,
     /// The sync handler of this node.
-    pub sync: OnceCell<Arc<Sync<S>>>,
+    pub sync: OnceCell<Arc<Sync>>,
+    pub storage: DynStorage,
```

Comment:
This is already available through `sync`; is it added to reduce indirection?

Reply:
it's just cleaner/easier to access. i plan to remove `OnceCell` from `sync` -- but trying to avoid making unnecessary sweeping changes in this PR more than i already have.